### PR TITLE
fix: undocumented args for torch frontend- added `axis` support for `stack`, and `value` support for `scatter_` + extended tests

### PR DESCRIPTION
**Pull request by mattbarrett98**
This adds support for behaviour which works for native torch i.e.  `torch.stack` allows the `dim` argument to also be passed using keyword `axis`, although it isn't in the documentation it does support this and so occasionally pops up in botorch.

### CODE REVIEW

**Comment by vedpatwardhan** [comment]
Oh sorry, just noticed that `test_torch_scatter_` seems to be failing in the [CI](https://github.com/unifyai/ivy/actions/runs/7873981485/job/21484066426?pr=28252#step:4:5447). Could you please confirm if the failures are related to the changes made? Thanks!

**Comment by hmahmood24** [comment]
> This adds support for behaviour which works for native torch i.e. `torch.stack` allows the `dim` argument to also be passed using keyword `axis`, although it isn't in the documentation it does support this and so occasionally pops up in botorch.

@mattbarrett98 There's a [numpy_to_torch_style_args](https://github.com/unifyai/ivy/blob/8dcc33b895240395686db165c710ac31708aa691/ivy/functional/frontends/torch/func_wrapper.py#L154) decorator in the `torch` frontend for this purpose I suppose
